src/lib/Scenes/Artwork/Components/CommercialInformation.tsx

```diff
@@ -71,12 +80,73 @@ export class CommercialInformation extends React.Component<CommercialInformation
     )
   }
 
+  componentDidMount = () => {
+    const { artwork } = this.props
+    const { sale } = artwork
+
+    if (sale && sale.isAuction) {
+      // Use local state to check auction status to update view on changes
+      if (artwork.sale.isClosed) {
+        this.setState({
+          auctionState: "hasEnded",
+        })
+        return
+      } else if (artwork.sale.isLiveOpen) {
+        this.setState({
+          auctionState: "isLive",
+        })
+      } else if (artwork.sale.isPreview) {
+        this.setState({
+          auctionState: "isPreview",
+        })
+      } else if (DateTime.local() > DateTime.fromISO(sale.startAt)) {
+        this.setState({
+          auctionState: "hasStarted",
+        })
+      }
+
+      // Set intervals to check for state changes
```

Comment:
Just noting that the auction timer employs a similar functionality w.r.t. keeping track of time and changing state accordingly.

Would it be worth trying to combine those functionalities or do you feel that they are separate enough? Under the hood the auction should be in the same state either way, so conceptually it seems they could be shared. But from a code organization perspective this might be better?

Curious what you think.

Reply:
I was thinking this too, I just removed the interval from the countdown timer component.